keras_cv/models/classification/image_classifier.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Image classifier model using pooling and dense layers."""
+
+import copy
+
+from keras import layers
+from tensorflow import keras
+
+from keras_cv.models.backbones.backbone_presets import backbone_presets
+from keras_cv.models.backbones.backbone_presets import (
+    backbone_presets_with_weights,
+)
+from keras_cv.models.classification.image_classifier_presets import (
+    classifier_presets,
+)
+from keras_cv.models.task import Task
+from keras_cv.utils.python_utils import classproperty
+
+
+@keras.utils.register_keras_serializable(package="keras_cv.models")
+class ImageClassifier(Task):
+    """Image classifier with pooling and dense layer prediction head.
+
+    Args:
+        backbone: `keras.Model` instance, the backbone architecture of the
+            classifier called on the inputs.
+        num_classes: int, number of classes to predict.
+        activation: A `str` or callable. The activation function to
+            use on the Dense layer. Set `classifier_activation=None` to return
+            the logits of the "top" layer.
+
+    Example:
+    ```python
+    input_data = tf.ones(shape=(8, 224, 224, 3))
+
+    # Pretrained classifier (e.g., for imagenet categories)
+    model = keras_cv.models.ImageClassifier.from_preset(
+        "resnet50_v2_imagenet_classifier",
+        activation="softmax",
+    )
+    output = model(input_data)
+
+    # Pretrained backbone
+    backbone = keras_cv.models.ResNet50V2Backbone.from_preset(
+        "resnet50_v2_imagenet",
+    )
+    model = keras_cv.models.ImageClassifier.from_preset(
```

Comment:
I don't understand this example. It says from_preset but there is no preset mentioned. Is it a typo ?